deploy/docker/notebook/base/base.yaml

```diff
@@ -421,26 +425,92 @@ addfiles:
+
+              def script_post_save(self, os_path):
+                  
+                  hookVar = hookVariables()
+
+                  os_path = str(Path().resolve()) + os_path
+
+                  if "jovyan" not in os_path:
+                      return
+
+                  if not (self.is_inside_path(os_path, hookVar.work_path) or self.is_inside_path(os_path, hookVar.shared_path)):
+                      return
+
+                  try:
+                      r = requests.post(hookVar.POLUS_NOTEBOOKS_HUB_API+"/file-data/new-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
+                  except Exception as e:
+                      if os.environ["POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED"] == "True":
+                          logging.getLogger().error(f"Failed to register {os_path} with Notebooks Hub", exc_info=e)
+
+              def script_post_rename(self, os_old_path, os_new_path):
+
+                  hookVar = hookVariables()
+                  
+                  os_old_path = str(Path().resolve()) + '/' + os_old_path
+                  os_new_path = str(Path().resolve()) + '/' + os_new_path
+
+                  if not (self.is_inside_path(os_new_path, hookVar.work_path) or self.is_inside_path(os_new_path, hookVar.shared_path)):
+                      return
+
+                  try:
+                      r = requests.post(hookVar.POLUS_NOTEBOOKS_HUB_API+"/file-data/rename-file", json={"user": os.getenv("JUPYTERHUB_USER"), "oldPath": os_old_path, "newPath": os_new_path}, timeout=1)
+                  except Exception as e:
+                      if os.environ["POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED"] == "True":
+                          logging.getLogger().error(f"Failed to change path from {os_old_path} to {os_new_path} with Notebooks Hub", exc_info=e)
+
+              def delete(self, os_path):
+                  returned = super(HookFileContentsManager, self).delete_file(os_path)
+                  self.script_post_delete(os_path)
+                  return returned
+
+              def save(self, model, os_path):
+                  returned = super(HookFileContentsManager, self).save(model, path=os_path)
+                  self.script_post_save(os_path)
+                  return returned
+
+              def rename(self, os_old_path, os_new_path):
+                  returned = super(HookFileContentsManager, self).rename_file(os_old_path, os_new_path)
+                  self.script_post_rename(os_old_path, os_new_path)
+                  return returned
+
+          #### Set contents manager class ####
+              
+          if os.environ["POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED"] == "True":
+              logging.getLogger("urllib3").propagate = True
+          else:
+              logging.getLogger("urllib3").propagate = False
+
+          c.NotebookApp.contents_manager_class = HookFileContentsManager
```

Comment:
I meant to use `POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED` not for controlling *logging*, but for enabling/disabling *file hooks*. I'm sorry if name is confusing, I could change it for something better. So, if the variable is `"True"`, I would like to enable `HookFileContentsManager` class as a Contents Manager Class, but do not do anything otherwise.